web/packages/teleport/src/Roles/RoleEditor/RoleEditor.tsx

```diff
@@ -84,34 +93,38 @@ export const RoleEditor = ({
   });
 
   // Defaults to yaml editor if the role could not be parsed.
-  const [selectedEditorTab, setSelectedEditorTab] = useState<EditorTab>(() =>
-    standardModel.roleModel.requiresReset ? EditorTab.Yaml : EditorTab.Standard
-  );
+  const [selectedEditorTab, setSelectedEditorTab] = useState<EditorTab>(() => {
+    const { roleModel } = standardModel;
+    return roleModel === undefined || roleModel.requiresReset
+      ? EditorTab.Yaml
+      : EditorTab.Standard;
+  });
 
   // Converts YAML representation to a standard editor model.
   const [parseAttempt, parseYaml] = useAsync(async () => {
-    const parsedRole = await yamlService.parse<Role>(
-      YamlSupportedResourceKind.Role,
-      {
-        yaml: yamlModel.content,
-      }
-    );
-    return roleToRoleEditorModel(parsedRole, originalRole?.object);
+    try {
+      const parsedRole = await yamlModelToRole(yamlModel);
+      return roleToRoleEditorModel(parsedRole, originalRole?.object);
+    } catch (err) {
+      throw new Error('Unable to load role into the standard editor', {
+        cause: err,
+      });
+    }
   });
 
   // The standard editor will automatically preview the changes based on state updates
   // but the yaml editor needs to be told when to update (the preview button)
-  const handleYamlPreview = useCallback(async () => {
```

Comment:
Nit, if it was important for `handleYamlPreview` to have a stable identity between renders before, then it should continue to use `useCallback`. `useAsync` itself cannot make the received callback stable.

Reply:
Good catch.